packages/react/src/primitives/shared/constants.ts

```diff
@@ -1,113 +1,504 @@
-  Flex = 'amplify-flex',
-  Grid = 'amplify-grid',
-  Heading = 'amplify-heading',
-  Icon = 'amplify-icon',
-  Image = 'amplify-image',
-  Input = 'amplify-input',
-  Label = 'amplify-label',
-  Link = 'amplify-link',
-  Loader = 'amplify-loader',
-  MenuContent = 'amplify-menu-content',
-  MenuItem = 'amplify-menu-content__item',
-  MenuTrigger = 'amplify-menu-trigger',
-  Pagination = 'amplify-pagination',
-  PaginationItemButton = 'amplify-pagination__item-button',
-  PaginationItemCurrent = 'amplify-pagination__item-current',
-  PaginationItemEllipsis = 'amplify-pagination__item-ellipsis',
-  PasswordField = 'amplify-passwordfield',
-  PhoneNumberField = 'amplify-phonenumberfield',
-  Placeholder = 'amplify-placeholder',
-  Radio = 'amplify-radio',
-  RadioButton = 'amplify-radio__button',
-  RadioInput = 'amplify-radio__input',
-  RadioLabel = 'amplify-radio__label',
-  RadioGroupField = 'amplify-radiogroupfield',
-  RadioGroup = 'amplify-radiogroup',
-  Rating = 'amplify-rating',
-  ScrollView = 'amplify-scrollview',
-  Select = 'amplify-select',
-  SelectField = 'amplify-selectfield',
-  SearchField = 'amplify-searchfield',
-  SearchFieldClear = 'amplify-searchfield__clear',
-  SearchFieldSearch = 'amplify-searchfield__search',
-  SelectWrapper = 'amplify-select__wrapper',
-  SelectIconWrapper = 'amplify-select__icon-wrapper',
-  SliderField = 'amplify-sliderfield',
-  SliderFieldGroup = 'amplify-sliderfield__group',
-  SliderFieldLabel = 'amplify-sliderfield__label',
-  SliderFieldRange = 'amplify-sliderfield__range',
-  SliderFieldRoot = 'amplify-sliderfield__root',
-  SliderFieldThumb = 'amplify-sliderfield__thumb',
-  SliderFieldTrack = 'amplify-sliderfield__track',
-  StepperField = 'amplify-stepperfield',
-  StepperFieldButtonDecrease = 'amplify-stepperfield__button--decrease',
-  StepperFieldButtonIncrease = 'amplify-stepperfield__button--increase',
-  StepperFieldInput = 'amplify-stepperfield__input',
-  SwitchField = 'amplify-switchfield',
-  SwitchLabel = 'amplify-switch-label',
-  SwitchThumb = 'amplify-switch-thumb',
-  SwitchTrack = 'amplify-switch-track',
-  SwitchWrapper = 'amplify-switch__wrapper',
-  Table = 'amplify-table',
-  TableCaption = 'amplify-table__caption',
-  TableBody = 'amplify-table__body',
-  TableTd = 'amplify-table__td',
-  TableTh = 'amplify-table__th',
-  TableFoot = 'amplify-table__foot',
-  TableHead = 'amplify-table__head',
-  TableRow = 'amplify-table__row',
-  Tabs = 'amplify-tabs',
-  TabItems = 'amplify-tabs-item',
-  Text = 'amplify-text',
-  Textarea = 'amplify-textarea',
-  TextField = 'amplify-textfield',
-  TextAreaField = 'amplify-textareafield',
-  ToggleButton = 'amplify-togglebutton',
-  ToggleButtonGroup = 'amplify-togglebuttongroup',
-  VisuallyHidden = 'amplify-visually-hidden',
-}
+export const ComponentClassObject = {
+  Alert: {
+    className: 'amplify-alert',
+    components: ['Alert'],
+    description: 'Top level element that wraps the Alert primitive',
```

Comment:
I don't have a better solution, what are people's thoughts about using a JS object here? This would be included in customers' bundles since it is a JS object as opposed to a TS enum. I think this is probably fine for now, its not too much extra JS to ship and it makes it much easier to document. We could think about using JSDoc/TSDoc comments for this depending on the investigation into generating documentation from TS/doc comments. I don't think it is blocking and it is not a 1-way door because customers shouldn't be using this object anyways.

Reply:
Good point. I agree that it's probably fine for now though.